cloudinit/sources/DataSourceHetzner.py

```diff
@@ -69,14 +75,18 @@ def get_data(self):
         # hostname is name provided by user at launch.  The API enforces it is
         # a valid hostname, but it is not guaranteed to be resolvable in dns or
         # fully qualified.
-        self.metadata['instance-id'] = md['instance-id']
+        self.metadata['instance-id'] = md.get('instance-id', serial)
```

Comment:
I wouldnt bother with this.
I'd rather check here that `md['instance-id'] == serial` and raise a runtime error.

But unless you a valid case where there may not be an instance-id in the metadata service (but there *is* a 'hostname' and we have an otherwise-functioning metadata service) then this is just confusing. and results in "sometimes i get the value from smbios and sometimes it comes from the MD".

Reply:
Good point. If both don't match, something is definitely wrong.